cookbooks/swift/recipes/setup.rb

```diff
@@ -54,92 +47,86 @@
   action [:disable, :stop]
 end
 
-# packages
+# system packages
 required_packages = [
   "libssl-dev", # libssl-dev is required for building wheels from the cryptography package in swift.
-  "curl", "gcc", "memcached", "rsync", "sqlite3", "xfsprogs", "git-core", "build-essential",
+  "curl", "gcc", "memcached", "rsync", "sqlite3", "xfsprogs", "git", "build-essential",
   "libffi-dev",  "libxml2-dev", "libxml2", "libxslt1-dev", "zlib1g-dev", "autoconf", "libtool",
   "haproxy", "docker-compose", "rclone",
 ]
 
-if node['platform_version'] == '22.04'
-  required_packages += [
-    "python2-dev", "python2", "python3", "python3-dev",
-    "python3.7", "python3.7-dev", "python3.7-distutils",
-    "python3.8", "python3.8-dev", "python3.8-distutils",
-    "python3.9", "python3.9-dev", "python3.9-distutils",
-  ]
-else
+# common python versions
+required_packages += [
+  # most of time these come from deadsnakes
+  "python3.7", "python3.7-distutils",
+  "python3.8", "python3.8-distutils",
+  "python3.9", "python3.9-distutils",
+  "python3.10",
+  "python3.11",
+  "python3.12",
+  "python3.13",  # edge of technology!
+  # python3 will be redundant with distro version, -dev is needed for pyeclib
+  "python3", "python3-dev",
+]
+
+# only focal has the *really* old py3
+if node['platform_version'].to_i <= 20 then
   required_packages += [
-    "python-dev",
-    "python3.6", "python3.6-dev", "python3.7", "python3.7-dev",
-    "python3.8", "python3.8-dev",
+    "python3.6", "python3.6-distutils",
   ]
 end
 
-extra_packages = node['extra_packages']
-(required_packages + extra_packages).each do |pkg|
-  package pkg do
-    action :install
-  end
-end
-
-# no-no packages (PIP is the bomb, system packages are OLD SKOOL)
+# no-no packages (PIP rules this vm, most system packages are all out-of-date anyway)
 unrequired_packages = [
-  "python-requests",  "python-six", "python-urllib3",
-  "python-pbr", "python-pip",
-  "python3-requests",  "python3-six", "python3-urllib3",
-  "python3-pbr", "python3-pip",
+  "python3-pip", "python3-pbr", "python3-setuptools",
+  "python3-openssl", "python3-certifi",
+  "python3-requests",  "python3-urllib3",
```

Comment:
Interesting -- so `setuptools`, `openssl`, and `certifi` are new entries here...

I'm rather shocked that any distro would have `certifi` terribly out of date. I'm guessing `openssl` is because of the trouble you were having with cryptography/pyopenssl. What's the deal with `setuptools`?

Reply:
you know honestly we could try trimming down this list and running the new `test.sh` script - I'm not 100% all of these are required some of it was just being defensive or flailing when I was troubleshooting focal.

I do think having a fairly anemic python environment on the system increases the odds of success.

Reply:
Tru dat